shard/committee/assignment.go

```diff
@@ -131,47 +140,156 @@ func with400Stakers(
 		stakers[i] = validator
 	}
 
+	for i := range stakers {
+		staker := stakers[i]
+		stakers[i].Stake = new(big.Int).Div(
+			staker.Stake, big.NewInt(int64(len(staker.SlotPubKeys))),
+		)
+	}
+
+	unsortedStakes := make([]int, len(stakers))
+	eposStakes := make([]*big.Int, len(stakers))
+
+	for i, j := range stakers {
+		unsortedStakes[i] = int(j.Stake.Int64())
+		eposStakes[i] = j.Stake
+	}
+
+	s3 := effective.Apply(eposStakes)
```

Comment:
The Epos calculation should apply on the individual slot level. Not on the validator level.

Reply:
I thought that is taken into account by: https://github.com/harmony-one/harmony/pull/1826/files#diff-5306cae6810bfd2a339378a4b76c1406R145

Reply:
actually just realized, my scheme doesn't work because it assume linearity, but epos is non-linear, so will change. (I initially thought of this way as an optimization on memory allocation)